### PR TITLE
Documentation for KReference.Group feature

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

## Proposed Changes 

- Documentation for the experimental feature https://github.com/knative/eventing/issues/5086

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
/hold

This is ready for review but the approval requires the approval of https://github.com/knative/eventing/pull/5440 first

**Review by rhuss — COMMENTED** [review]
Looks good in general but could elaborate a bit why to prefer `group:` over `apiVersion:` like you do in the description of the implementation PR ?

**Comment by slinkydeveloper** [comment]
@rhuss last commit should explain the "why", lmk how it sounds

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:80]:
```suggestion
!!! note
    At the moment this feature is implemented only for `Subscription.Spec.Subscriber.Ref`.
```

We tend not to mention future plans as best practice in docs

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:78]:
```suggestion
With this feature you can allow Knative to resolve the full `APIVersion` and further upgrades, deprecations, and removals of the referred CRD without affecting existing resources.
```

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:60]:
```suggestion
When using the `KReference` type to refer to another Knative resource, you can just specify the API `group` of the resource, instead of the full `APIVersion`.
```

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:62]:
```suggestion
For example, in order to refer to an `InMemoryChannel`, instead of using the following spec:
```

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:70]:
```suggestion
You can use the following:
```

**Review by abrennan89 — COMMENTED** [review]
Added some suggestions

File: docs/eventing/experimental-features.md
Comment (slinkydeveloper) [reply docs/eventing/experimental-features.md:80]: There is a plan to follow for this feature which includes using that field pretty much everywhere in our apis... I do wonder, maybe should I link the issue somewhere here in the docs?

Comment (abrennan89) [reply docs/eventing/experimental-features.md]: @slinkydeveloper please open docs issues for any future additions so we can track it and make sure it gets added rather than mentioning it here 🙂

**Review by abrennan89 — APPROVED** [review]
/lgtm

File: docs/eventing/experimental-features.md
Comment (slinkydeveloper) [reply docs/eventing/experimental-features.md]: I meant adding a link here in this paragraph, let me add it so you see what i mean

Comment (abrennan89) [reply docs/eventing/experimental-features.md]: Oh, I see. No, please don't link to upcoming issues in the docs, it's the same really as mentioning future plans and we should avoid that.

Comment (slinkydeveloper) [reply docs/eventing/experimental-features.md]: Check it out now

File: docs/eventing/experimental-features.md
Comment (abrennan89) [inline docs/eventing/experimental-features.md:60]: If the feature is implemented as it's documented for now I don't think there's a need to add this. If it's not implemented, the docs shouldn't be added until it is.

Comment (slinkydeveloper) [reply docs/eventing/experimental-features.md:60]: But this is an experimental feature, that is a feature which development is ongoing. Hence, to me it seems reasonable to link where the development is tracked. As an example, the Rust community does the same: https://doc.rust-lang.org/std/alloc/struct.Layout.html#method.for_value_raw

Comment (abrennan89) [reply docs/eventing/experimental-features.md:60]: OK

**Comment by slinkydeveloper** [comment]
@abrennan89 can you lgtm it again? :)

**Pull request merged**
